platform_plugin_turnitin/tasks.py

```diff
@@ -26,84 +25,87 @@
 
 
 @shared_task
-def ora_submission_created_task(submission_id: str, file_downloads: List[dict]) -> None:
+def ora_submission_created_task(
+    submission_uuid: str,
+    anonymous_user_id: str,
+    parts: List[dict],
+    file_names: List[str],
+    file_urls: List[str],
+) -> None:
     """
     Task to handle the creation of a new ora submission.
 
     Args:
-        submission_id (str): The ORA submission ID.
-        file_downloads (List[dict]): The list of file downloads.
+        submission_uuid (str): The ORA submission UUID.
+        anonymous_user_id (str): The anonymous user ID.
+        parts (List[dict]): The parts of the submission with the answers.
+        file_names (List[str]): The list of file names.
+        file_urls (List[str]): The list of file URLs.
     """
-    submission_data = dict(submissions_api.get_submission_and_student(submission_id))
-    user = user_by_anonymous_id(submission_data["student_item"]["student_id"])
+    user = user_by_anonymous_id(anonymous_user_id)
 
-    send_text_to_turnitin(submission_id, user, submission_data["answer"])
-    send_uploaded_files_to_turnitin(submission_id, user, file_downloads)
+    send_text_to_turnitin(submission_uuid, user, parts)
+    send_uploaded_files_to_turnitin(submission_uuid, user, file_names, file_urls)
 
     for _ in range(MAX_REQUEST_RETRIES):
-        if is_submission_complete(submission_id, user):
-            generate_similarity_report(submission_id, user)
+        if is_submission_complete(submission_uuid, user):
+            generate_similarity_report(submission_uuid, user)
             break
         sleep(SECONDS_TO_WAIT_BETWEEN_RETRIES)
 
 
-def send_text_to_turnitin(submission_id: str, user, answer: dict) -> None:
+def send_text_to_turnitin(ora_submission_uuid: str, user, parts: List[dict]) -> None:
     """
     Task to send text to Turnitin.
 
     Args:
-        submission_id (str): The ORA submission ID.
+        ora_submission_uuid (str): The ORA submission UUID.
         user (User): The user who made the submission.
-        answer (dict): The answer of the submission.
+        parts (List[dict]): The answer of the submission.
     """
-    for part in answer["parts"]:
+    for part in parts:
         text_content = part.get("text").encode("utf-8")
-        send_file_to_turnitin(
-            submission_id, user, text_content, "Students' Text Response.txt"
-        )
+        send_file_to_turnitin(ora_submission_uuid, user, text_content, "Students' Text Response.txt")
```

Comment:
could we add an index to Students' Text Response.txt? So it's not called the same for all parts? Or use a part of the text content as the name?